evalml/pipelines/components/transformers/encoders/target_encoder.py

```diff
@@ -68,3 +68,6 @@ def get_feature_names(self):
             np.array: The feature names after encoding
         """
         return self._component_obj.get_feature_names()
+
+    def _get_feature_provenance(self):
```

Comment:
There is no uniform logic yet! For the target encoder we get a `ComponentNotYetFittedError` but for the date time featurizer we get a stack trace lol.

I think that's ok for now since we're only using this internally. If we ever decide to make it public we should definitely modify the metaclass to add some guardrails.